### PR TITLE
feat: Implement hiding input values

### PR DESCRIPTION
**Pull request by martinalbert**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the documentation accordingly. If you are changing **code related to user secrets** you need to really make sure that [security documentation](SECURITY.md) is correct.
- [x] I have read the **CONTRIBUTION_GUIDE** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by Jakub-Vacek** [comment]
I think this branch also contains changes from https://github.com/superfaceai/testing/pull/21/files so it is kinda hard to review :/

File: README.md [hunk 1]
```diff
@@ -192,7 +192,31 @@ superface.run(
...
+    },
+  },
+  {
+    // value found have to be one of following: string, number or boolean
+    hideInput: ['auth.value', 'auth.password']
+  }
+);
+```
+
+You can also enter your own processing functions along side `processRecordings` parameter. Both have same function signature and are called either before load or before save of recordings.
```
Comment (freaz) [inline README.md:219]: I'd add example as well.

File: src/superface-test.test.ts [hunk 2]
```diff
@@ -275,7 +280,109 @@ describe('SuperfaceTest', () => {
...
+
+        await superfaceTest.run(
+          { input: { auth: { token, refresh } } },
+          { hideInput: ['auth.token', 'auth.refresh'] }
+        );
+
+        expect(playSpy).toHaveBeenCalledTimes(1);
+        expect(endRecSpy).toHaveBeenCalledTimes(1);
+
+        expect(writeRecordingsSpy).toHaveBeenCalledTimes(1);
```
Comment (freaz) [inline src/superface-test.test.ts]: Are this assertions relevant to test case? I'd assume there is another testcase ensuring play and endRec is called. Write I guess is to be sure it is called exactly once?

File: src/nock.utils.test.ts [hunk 3]
```diff
@@ -64,18 +66,19 @@ describe('nock utils', () => {
...
           path: '/get?text=123',
           method: 'GET',
           status: 200,
-          rawHeaders: ['api_key', HIDDEN_CREDENTIALS_PLACEHOLDER],
+          rawHeaders: ['api_key', TMP_PLACEHOLDER],
         });
       });
 
-      it('removes apikey from body', () => {
+      it('replaces apikey from body', () => {
```
Comment (freaz) [inline src/nock.utils.test.ts]:
```suggestion
      it('replaces apikey in body', () => {
```

File: src/nock.utils.test.ts [hunk 4]
```diff
@@ -40,12 +42,12 @@ describe('nock utils', () => {
...
           status: 200,
           reqheaders: {
-            ['api_key']: HIDDEN_CREDENTIALS_PLACEHOLDER,
+            ['api_key']: TMP_PLACEHOLDER,
           },
         });
       });
 
-      it('removes apikey from raw headers', () => {
+      it('replaces apikey from raw headers', () => {
```
Comment (freaz) [inline src/nock.utils.test.ts]:
```suggestion
      it('replaces apikey in raw headers', () => {
```

File: src/nock.utils.test.ts [hunk 5]
```diff
@@ -2,16 +2,17 @@ import { ApiKeyPlacement, HttpScheme, SecurityType } from '@superfaceai/ast';
...
 
+const TMP_PLACEHOLDER = 'placeholder';
+
 describe('nock utils', () => {
-  describe('removeCredentials', () => {
-    describe('when removing apikey', () => {
-      it('removes apikey from header', () => {
+  describe('replaceCredentials', () => {
+    describe('when replacing apikey', () => {
+      it('replaces apikey from header', () => {
```
Comment (freaz) [inline src/nock.utils.test.ts]:
```suggestion
      it('replaces apikey in header', () => {
```

**Review by freaz — APPROVED** [review]
Looks good.

Found few nitpicks, I hope didn't miss anything important.

File: README.md (hunk 1 above)
Comment (martinalbert) [reply README.md:219]: It's badly displayed in changed files, but there are some example bellow. Let me know if it's enough.

File: src/superface-test.test.ts (hunk 2 above)
Comment (martinalbert) [reply src/superface-test.test.ts]: True, I'll remove those that are tested before. And yes, write is done only once per `run()` with all definitions in record mode after perform is executed.
